src/GoogleBooks.php

```diff
@@ -119,24 +118,34 @@ public function getItem($path)
         return $this->raw($path);
     }
 
-    public function listItems($endpoint, $params = [])
+    
+    public function chunk(\Generator $generator, $chunkInt)
     {
-        $params['maxResults'] = $this->batchSize;
+        $chunk = [];
+        foreach ($generator as $gen) {
+            $chunk[] = $generator->current();
+        }
+        return array_chunk($chunk, $chunkInt);
+    }
 
+    public function listItems($endpoint, $params = [])
+    {
         $i = 0;
+        $return = [];
         while (true) {
             $n = $i % $this->batchSize;
+            $params['maxResults'] = $this->batchSize;
+            
```

Comment:
Perhaps we should add a `.editorconfig` file? I prefer having "trim_trailing_whitespace: true"

Reply:
haven't used a `.editorconfig` file. can you elaborate more?

Reply:
See http://editorconfig.org/ . Quite useful to keep things consistent when people use different editors (and platforms). I suggest something like this:

```
root = true

[*]
end_of_line = lf
insert_final_newline = true

[*.{php}]
charset = utf-8
indent_style = space
indent_size = 4
```